src/c/adaptors/rocksdb_impl.hpp

```diff
@@ -91,7 +105,14 @@ struct RocksDBStorage : public Storage {
              const std::vector<uint8_t>& value) override {
     rocksdb::Slice key_slice((char*)key.data(), key.size());
     rocksdb::Slice value_slice((char*)value.data(), value.size());
-    db_->Put(write_options_, key_slice, value_slice);
+
+    rocksdb::Status status = db_->Put(write_options_, key_slice, value_slice);
+    if (!status.ok()) {
+      throw altintegration::StorageWriteException(
+          fmt::format("failed to write into the storage, key: %s, value: %s",
+                      altintegration::HexStr(key),
+                      altintegration::HexStr(value)));
```

Comment:
instead of this msg, better format `status` - it will contain reason why this failed